lib/payment/payment_service.dart

```diff
@@ -0,0 +1,154 @@
+import 'package:coffeecard/data/repositories/v2/purchase_repository.dart';
+import 'package:coffeecard/generated/api/coffeecard_api_v2.swagger.swagger.dart';
+import 'package:coffeecard/service_locator.dart';
+import 'package:coffeecard/widgets/popup_card.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter/services.dart';
+
+enum InternalPaymentType {
+  mobilePay,
+  applePay,
+}
+
+abstract class PaymentService {
+  factory PaymentService(InternalPaymentType type, BuildContext context) {
+    switch (type) {
+      case InternalPaymentType.mobilePay:
+        return MobilePayService(context);
+      case InternalPaymentType.applePay:
+        throw UnimplementedError();
+    }
+  }
+
+  Future<Payment> initPurchase(int productId);
+  void invokeMobilePay(String paymentId, int price);
+  Future<PaymentStatus> verifyPurchaseOrRetry(int purchaseId);
```

Comment:
I think these abstract methods are to tightly coupled to MobilePay if the PaymentService should be an abstract interface for possibly Multiple Payment Types

Suggested methods 
```dart
// Calls API, parse deeplink MobilePay, checks Payment status and returns successful or error. I got the idea that we could do it more functional with Either instead of throwing exceptions.
Either<SuccessfulType, ErrorType> purchase(int productId) 

// Gives user a current status of purchase, used to re-init purchase or other control flow
Either<SuccessfulType, ErrorType> verifyPurchase(some relevant identifier)
```